windowlist@cobinja.de/files/windowlist@cobinja.de/6.4/applet.js

```diff
@@ -0,0 +1,2344 @@
+    return true;
+  }
+  let transientHasFocus = false;
+  metaWindow.foreach_transient(function(transient) {
+    if (transient.appears_focused) {
+      transientHasFocus = true;
+      return false;
+    }
+    return true;
+  });
+  return transientHasFocus;
+}
+
+function compareObject(x, y) {
+  // mimic non-extisting logical xor
+  // to determine if one of the
+  // parameters is undefined and the other one is not
+  if (!(x == undefined) != !(y == undefined)) {
+    return false;
+  }
+  if (x === y) {
+    return true;
+  }
+  
+  if (!(x instanceof Object) || !(y instanceof Object)) {
+    return false;
+  }
+  
+  if (Object.getPrototypeOf(x) !== Object.getPrototypeOf(y)) {
+    return false;
+  }
+  
+  let keysX = Object.keys(x);
+  let keysY = Object.keys(y);
+  
+  if (keysX.length != keysY.length) {
+    return false;
+  }
+  
+  for (let i = 0; i < keysX.length; i++) {
+    let key = keysX[i];
+    if (!y.hasOwnProperty(key)) {
+      return false;
+    }
+    if (!compareObject(x[key], y[key])) {
+      return false;
+    }
+  }
+  
+  return true;
+}
+
+function sign(p1, p2, p3) {
+  return (p1.x - p3.x) * (p2.y - p3.y) - (p2.x - p3.x) * (p1.y - p3.y);
+}
+
+function pointInTriangle(pt, v1, v2, v3) {
+  let b1 = sign(pt, v1, v2) < 0.0;
+  let b2 = sign(pt, v2, v3) < 0.0;
+  let b3 = sign(pt, v3, v1) < 0.0;
+  return ((b1 == b2) && (b2 == b3));
+}
+
+function resizeActor(actor, time, toWidth) {
+  Tweener.addTween(actor, {
+    natural_width: toWidth,
+    time: time * 0.001,
+    transition: "easeInOutQuad"
+  });
+}
+
+function getOverheadSize(actor) {
+  if (actor == null) {
+    return null;
+  }
+  let height = 0;
+  let width = 0;
+  let themeNode = actor.get_theme_node();
+  
+  width = themeNode.get_padding(St.Side.LEFT);
+  width += themeNode.get_padding(St.Side.RIGHT);
+  width += themeNode.get_border_width(St.Side.LEFT);
+  width += themeNode.get_border_width(St.Side.RIGHT);
+  
+  height = themeNode.get_padding(St.Side.TOP);
+  height += themeNode.get_padding(St.Side.BOTTOM);
+  height += themeNode.get_border_width(St.Side.TOP);
+  height += themeNode.get_border_width(St.Side.BOTTOM);
+  
+  width += themeNode.get_margin(St.Side.LEFT);
+  width += themeNode.get_margin(St.Side.RIGHT);
+  height += themeNode.get_margin(St.Side.TOP);
+  height += themeNode.get_margin(St.Side.BOTTOM);
+  
+  return [width, height];
+}
+
+function getMonitors() {
+  let result = [];
+
+  try {
+    let gdkScreen = Gdk.Screen.get_default();
+    // let screen = CinnamonDesktop.RRScreen.new(gdkScreen);
+    // let currentConfig = CinnamonDesktop.RRConfig.new_current(screen);
+    // let outputInfos = currentConfig.get_outputs();
+
+    // for (let index = 0; index < outputInfos.length; index++) {
+    //   let output = outputInfos[index];
+    //   if (output.is_active()) {
+    //     result.push(output.get_display_name());
+    //   }
+    // }
+  } catch (err) {
+    return [];
+  }
+
+  return result;
+}
```

Comment:
CassiaWindowList will also need this to avoid issues with Cinnamon 6.4.

cc @klangman